HW2/P2/parallel_vector.pyx

```diff
@@ -81,10 +81,17 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
    ##########
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+           for idx in prange(src.shape[0], num_threads = 4):
                if counts[src[idx]] > 0:
```

Comment:
acquiring the lock here may be too late, and counts[src[idx]] may have been invalidated since checking the condition (e.g., it is no longer > 0)